### PR TITLE
react-noop-renderer: convert vars into let/const

### PR DESCRIPTION
**Pull request by raphamorim**
ref: #11730 
last one @gaearon, I promise 😆

### CODE REVIEW

**Comment by gaearon** [comment]
Was `react-dom` fully converted? Pretty sure I saw some vars there..

**Comment by blling** [comment]
How to prevent developer from use `var` again?

**Comment by gaearon** [comment]
We could lint against it but this is a bit tricky because some files *have to* use `var`. I think ideally we should have a way to override lint configuration for a subset of files. We have the same problem in Prettier config (but solved it there).

**Comment by raphamorim** [comment]
Hm. React-DOM, I keep some `vars` because of some SSR tests are using `DEV` blocks (I believe this is the reason)

**Comment by gaearon** [comment]
Can you explain that in more detail?

**Comment by raphamorim** [comment]
One sec. I'll give a example

File: packages/react-dom/src/server/ReactPartialRenderer.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ var toArray = ((React.Children.toArray: any): toArrayType);
 var getStackAddendum = emptyFunction.thatReturns('');
 
 if (__DEV__) {
-  var validatePropertiesInDevelopment = function(type, props) {
+  let validatePropertiesInDevelopment = function(type, props) {
```
Comment (raphamorim) [inline packages/react-dom/src/server/ReactPartialRenderer.js]: Hm, seems to be a scope issue.
In this case I simple changed it and returns erros like `ReferenceError: validatePropertiesInDevelopment is not defined` on tests.

Comment (gaearon) [reply packages/react-dom/src/server/ReactPartialRenderer.js]: Yep—you can define it outside the dev block.

Comment (gaearon) [reply packages/react-dom/src/server/ReactPartialRenderer.js]: (but assign it inside)

Comment (gaearon) [reply packages/react-dom/src/server/ReactPartialRenderer.js]: Maybe `let validatePropertiesInDevelopment = () => {};` would be the best default value, and then in the block you reassign.

Comment (raphamorim) [reply packages/react-dom/src/server/ReactPartialRenderer.js]: ok, I'll fix for `react-dom`, and see if forgot some var then. I'll add on this PR.

Comment (raphamorim) [reply packages/react-dom/src/server/ReactPartialRenderer.js]: It's fine for you?

Comment (raphamorim) [reply packages/react-dom/src/server/ReactPartialRenderer.js]: I think it's better I close this PR and send one with another branch name :P

**Pull request closed without merging**